### PR TITLE
Update Forms block smoke testing for broader coverage

### PR DESCRIPTION
**Pull request by dpasque**
Related to #

## Proposed Changes

This reworks and greatly expands our Forms block smoke testing.

I've arranged three flows that I think together cover the majority of editor-focused Form cases:

1. The "All Fields" flow, where we build a form with literally every possible form field to make sure all are compatible with the editor.
2. The "Contact Form" flow, where we use the "Contact Form" block to test the pre-canned Form options that are presented to users as blocks in the block library.
3. The "Form Patterns" flow, where we launch the patterns modal from the "Form" block and build the form out of a pattern/

To make all of these play together nicely, we rely on label prefixing -- we label potentially conflicting fields with a prefix unique to the flow. This allows easy, a11y-friendly validation of all form field inputs on the published page! 🎉 

## Testing Instructions

- [x] Gutenberg tests pass
- [x] Jetpack Simple Mobile
- [x] Jetpack Simple Desktop
- [x] Jetpack Atomic

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts
Comment (dpasque) [inline packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: Without exact matching, we couldn't ever select "Form", because it always pulled in "Preformatted" 🤦

File: test/e2e/specs/blocks/shared/block-smoke-testing.ts
Comment (dpasque) [inline test/e2e/specs/blocks/shared/block-smoke-testing.ts:65]: This is a new change to the editor context -- we now pass in a locator (built from the block guid) that points to the parent element of the block that was just added! This should allow for blocks with similar elements to be more compatible within the same test suite.

File: packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts:15]: **question:non-blocking** - I went the other way and defined an additional parameter in the constructor `validationData` in some of the block tests I've written.
Example: https://github.com/Automattic/wp-calypso/pull/81065/files#diff-415c2abb9f3f69f190ceba55b72002f402ed52da4c321e0b10bb2e5ce3c2bac4R36


What do you think about clearly separating the `configurationData` and `validationData`?

File: packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts]: **nit:blocking** - we should use the following

```suggestion
		await context.page.waitForTimeout( 2 * 1000 );
```

File: packages/calypso-e2e/src/lib/blocks/block-flows/shared/block-helpers.ts
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/blocks/block-flows/shared/block-helpers.ts]: **suggestion:non-blocking** - can we consider moving the interface and type definitions up top, grouped together?

File: packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: This change _should_ not affect any existing flows, but I've triggered a build on Gutenberg Simple production to double check.

Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: Okay, so this does have some fallout. There are two issues:

- Pay with Pay**p**al block needs an uppercase P where bolded;
- Instagram has this (" Instagram") weird unicode character in front, which needs to be accounted for. 

image

Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: I filed an issue for the second one. https://github.com/Automattic/wp-calypso/issues/81135 and asked Jeremy for priority.

The first issue is easy to fix - I'll leave it up to you whether you want to make the one-line change in this PR or I can do it.

**Review by worldomonation — APPROVED** [review]
It's definitely a PR that's more on the complex side of things, but I can follow along and it appears to be a good approach to the problem!

File: packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: Thank you Edwin for testing that! 😄  I ended up not having time on Friday to test all the prior block flows with the change. 

> I filed an issue for the second one. https://github.com/Automattic/wp-calypso/issues/81135 and asked Jeremy for priority.

Looks like he already has a PR for this! 🎉  I'll wait to merge until this is merged.

> The first issue is easy to fix - I'll leave it up to you whether you want to make the one-line change in this PR or I can do it.

I saw you did this in your Ads PR (thank you), and I can either copy it here and/or just rebase before I merge. I'll make sure everything is green on this branch either way before deploying. 😄

File: packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/blocks/block-flows/form-patterns.ts:15]: Oh sure, that's a great idea! 😄 I'll pull the `otherExpectedFields` to a new interface 👍

**Comment by dpasque** [comment]
Pushed up the PR fixes -- still going to wait on https://github.com/Automattic/jetpack/pull/32705 before merging to make sure all the Gutenberg tests keep passing 👍

File: packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/editor-sidebar-block-inserter-component.ts:83]: My Ads PR requires this PR to be merged (I built on top of your changes) so if preferable, let's deploy yours first, and then immediately deploy my Ads PR after.

**Comment by jeherve** [comment]
> still going to wait on https://github.com/Automattic/jetpack/pull/32705 before merging to make sure all the Gutenberg tests keep passing 👍

It's merged now 👍

**Comment by worldomonation** [comment]
Spot-checked the editor for `jetpackAtomicWpBetaUser` and looks like the updated Jetpack build is deployed, so I think this branch is good to go. 

I'll try triggering new builds.

**Comment by dpasque** [comment]
All green now, let's merge! 🚀

**Pull request merged**
